### PR TITLE
Add ability to upload images via form-data

### PR DESCRIPTION
**Pull request by zachseidner1**
## Overview

Extend the upload route to accept images using form-data. 

## Changes Made

- Add `file_data` parameter for the util for updating images
- Add documentation for the `upload_image_helper` function
- Modify the original upload route to check for form-data.

## Test Coverage

I tested uploading a 24.9 MB image to the service with Postman and it worked fine. I tried uploading JPG and png image files through form data and they worked as expected. Also tested uploading images through base64 in the request body and it still worked.

## Screenshots

image

### CODE REVIEW

File: utils.py [hunk 1]
```diff
@@ -22,21 +22,48 @@ def failure_response(message, code=404):
...
+    if image_data:
+        mime_type = guess_type(image_data)[0]
+        ext = guess_extension(guess_type(image_data)[0])[1:]
+        if re.fullmatch(ALLOWED_MIME_TYPES_REGEX, mime_type) is None:
+            raise Exception(f"Extension {ext} not supported!")
+    else:
+        mime_type = guess_type(file_data.filename)[0]
+        ext = guess_extension(mime_type)[1:]
+        if re.fullmatch(ALLOWED_MIME_TYPES_REGEX, mime_type) is None:
+            raise Exception(f"Extension {ext} not supported!")
```
Comment (Archit404Error) [inline utils.py]: Note that this code is essentially duplicated since both branches have the same structure.

File: utils.py [hunk 2]
```diff
@@ -46,13 +73,21 @@ def upload_image_helper(image_data, bucket_name):
...
+            Body=file_data.read(),
+            ACL="public-read",
+        )
+    else:
+        client.put_object(
+            Bucket=bucket_name,
+            Key=img_filename,
+            Body=BytesIO(img_bytes),
+            ACL="public-read",
+        )
```
Comment (Archit404Error) [inline utils.py]: Also essentially duplicated code that can be factored out. I think a more intuitive way to structure this code would be to create a helper upload method and then simple methods for base64 and file upload that just pass necessary params into the helper

Would also make the function spec easier to understand

**Review by Archit404Error — APPROVED** [review]
I'm approving because this looks solid but I'd recommend making the refactor via the comments above to pull out duplicated code before merge. `upload_image_helper()` seems a bit convoluted and I'm guessing that's why things took a while to debug.

**Review by joycrwu — APPROVED** [review]
yayy!!

**Pull request merged**